### ISSUE TITLE
Cannot handle column names that match SQL keywords.

### ISSUE THREAD

**Issue by shuangye** [rbock/sqlpp11#186]
I have a column named "order", which is also a SQL keyword. When I execute a select, it throws an exception:
terminate called after throwing an instance of 'sqlpp::exception'
  what():  Sqlite3 error: Could not prepare statement: near "order": syntax error (statement was >>SELECT Areas.uuid,Areas.order FROM Areas<<
Putting the table name and column names in a quote (\') works fine: SELECT \`Areas\`.\`uuid\`,\`Areas\`.\`order\` FROM \`Areas\`.
But I have not find where the SQL statement is constructed in sqlpp11.

**Comment by rbock** [comment]
Hi,

Yes, as of now, column names are not quoted. You could change that in column.h (the serialization function is at the bottom). Want to experiment with that and maybe create a pull request?

Cheers,

Roland

**Comment by shuangye** [comment]
Thank you. I've added the quotes and tested. Pull request has been submitted.

**Comment by Erroneous1** [comment]
This has been discussed before with issue #152 . The issue with quoting all tables is that some databases do not support backticks to quote names. The advice is to put the backticks in your column definition.

**Comment by shuangye** [comment]
Right, double quotes works. 
```
CREATE TABLE Test("order" INT);
SELECT "order" FROM "Test";
```
Anyway, column.h needs to be modified to include such a mechanism to escape SQL keywords.

**Comment by rbock** [comment]
@Erroneous1 Thanks for the reminder! I need to add this to the documentation, including the link to stack SO. That is one awesome answer!

@shuangye Erroneous1 is right, the quoting should be done at the source, which is in the table definition (sorry for leading you towards column.h yesterday, obviously not thinking clearly). The question is how to do it? I would prefer not to have a long list of potential keywords (which might differ from database to database). Maybe quotes should be read from the DDL?

**Comment by Erroneous1** [comment]
According to the SO answer, ``` ` ```, `"`, and `[ ]` are the 3 ways to do it. It sounds like not all RDBMs support `"`, and it could be confused with text. A column could have a static constexpr enum or tag lets the serializer put quotes as either none (default), backtick, double_quote, or bracket.

Maybe something along these lines:
```c++
namespace quote_tag
{
struct none {};
struct backtick {};
struct double_quote {};
struct bracket {};
}

template<class Column, class = void_t<> >
struct requires_backtick : std::false_type {};

template<class Column>
struct requires_backtick<Column, void_t<std::enable_if<std::is_same<typename Column::quote_type,quote_tag::backtick>::value> > > : std::true_type {};

//... and similar for requires_double_quote, requires_bracket.
```

**Comment by rbock** [comment]
Sure, we could do something like that (I would probably write functions `quote_left` and `quote_right` to put the respective characters into the stream), but we could also simply take quotations from the DDL and write them into `_literal`.
This way, we would get the quotes as required by the database for the names that require quoting.

**Comment by rbock** [comment]
Merged into #199
